drivers/dma/dma_silabs_siwx91x.c

```diff
@@ -68,7 +70,7 @@ static int siwx91x_transfer_direction(uint32_t dir)
 		return TRANSFER_TO_OR_FROM_PER;
 	}
 
-	return -EINVAL;
+	return TRANSFER_DIR_INVALID;
```

Comment:
You still need to update the return value type of `siwx91x_transfer_direction()` from `int` to `enum dma_xfer_dir`